sympy/solvers/solveset.py

```diff
@@ -1332,6 +1338,83 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _is_bivariate(f, symbol):
+    r"""
+    Helper to check if the equation takes bivariate form.
+    If the equation has exactly two generators then we can
+    say it to be in bivariate form.
+    """
+    poly = f.as_poly()
+    gens = _filtered_gens(poly, symbol)
+
+    return len(gens) == 2
+
+
+def _solve_as_bivariate(lhs, rhs, symbol, domain):
+    r"""
+    Helper solver to solve equations of bivariate form.
+    """
+    result = ConditionSet(symbol, Eq(lhs - rhs, 0), domain)
+
+    poly = lhs.as_poly()
+    gens = _filtered_gens(poly, symbol)
+    gpu = bivariate_type(lhs - rhs, *gens)
+    if gpu is not None:
+        g, p, u = gpu
+        if g != lhs:
+            usol = _solveset(p, u, domain)
+            if not isinstance(usol, ConditionSet):
+                result = [_solveset(g - us, symbol, domain) for us in usol]
+    return result
+
+
+def _is_lambert(f, symbol):
+    r"""
+    Helper to check whether an equation is of lambert type.
+    Equations containing `Pow`, `log` or `exp` terms are currently checked
+    for having lambert solutions
```

Comment:
period at end of sentence

Reply:
Here, and throughout, I believe Lambert should be capitablized when refering to solutions or equations.